structuralcodes/sections/_generic.py

```diff
@@ -478,6 +581,233 @@ def calculate_moment_curvature(
 
         return res
 
+    def calculate_interaction_domain(
+        self, theta: float = 0, n_axial: int = 100
```

Comment:
Same as for the result dataclass, could we say `num_axial` to avoid confusion with the axial force?